rewrite-java/src/main/java/org/openrewrite/java/NoStaticImport.java

```diff
@@ -66,8 +66,17 @@ public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, Execu
                     if (m.getMethodType() != null) {
                         JavaType.FullyQualified receiverType = m.getMethodType().getDeclaringType();
 
-                        RemoveImport<ExecutionContext> op = new RemoveImport<>(receiverType.getFullyQualifiedName() + "." + method.getSimpleName(),
-                                true);
+                        // Do not replace if receiverType is the same as surrounding class
+                        if (getCursor().firstEnclosing(J.ClassDeclaration.class).getType().equals(receiverType)) {
```

Comment:
I suggest to also add a test with a nested type accessing a static member from the outer type and vice-versa.